### PR TITLE
build: define TZDIR for tzcode build

### PR DESCRIPTION
**Pull request by grafin**
NixOS (and probably some other distributives) places `zoneinfo` directory not to `/usr/share`, but rather in directory of a form `/nix/store/izida45xkdjkejekje8282hdhdh6q3-tzdata-2022a/share/zoneinfo/`. In this case NixOS sets `TZDIR` environment variable accordingly. Currently, in Tarantool we hardcode  `/usr/share/zoneinfo` as a location to use, disregarding `TZDIR` env variable.

This commit adds compile-time definition for `TZDIR` if such env variable defined, which allows to fix zoneinfo lookup in case of NixOS.

### CODE REVIEW

**Review by tsafin — APPROVED** [review]
LGTM

File: src/lib/tzcode/CMakeLists.txt [hunk 1]
```diff
@@ -13,6 +13,10 @@ set(TZDATA_FULL_TARGET ${TZCODE_SOURCE_DIR}/${TZDATA_TARGET})
 set(TZCODE_SRC ${PROJECT_SOURCE_DIR}/src/lib/tzcode)
 set(TZLUA_SRC ${PROJECT_SOURCE_DIR}/src/lua)
 
+if(DEFINED ENV{TZDIR})
```
Comment (ligurio) [inline src/lib/tzcode/CMakeLists.txt:16]: We can evaluate `ENV{TZDIR}` only once and then use variable with value. I'll not insist, it's up to you.

**Review by ligurio — APPROVED** [review]
LGTM (see a small comment inline)

**Review by kyukhin — APPROVED** [review]

**Comment by kyukhin** [comment]
Backported to 2.10 as well.

**Pull request merged**